pkg/daemon/ceph/client/mirror.go

```diff
@@ -136,7 +137,27 @@ func enablePoolMirroring(context *clusterd.Context, clusterInfo *ClusterInfo, po
 		return errors.Wrapf(err, "failed to get mirroring info for the pool %q", pool.Name)
 	}
 
-	if pool.Mirroring.Mode == mirrorInfo.Mode {
+	// List all the radosnamespace and get the mirroring mode for the implicit namespace
+	namespaces, err := context.RookClientset.CephV1().CephBlockPoolRadosNamespaces(clusterInfo.Namespace).List(clusterInfo.Context, metav1.ListOptions{})
+	if err != nil {
+		return errors.Wrapf(err, "failed to list namespace for pool %q", pool.Name)
+	}
+
+	implicitMirroringMode := ""
+	for _, namespace := range namespaces.Items {
```

Comment:
Looping through all radosnamespaces could be expensive if there are a lot of them. Most clusters won't have a lot, but what if they do? 

What about a different approach where we check for this case in the rados namespace controller? The rados namespace controller can easily look up its single parent block pool for the implicit namespace. If the pool has a different mirroring mode than the rados namespace, then it can be an error on the rados namespace reconcile. But if the implicit namespace and the parent pool have the same mirroring setting, then there is no need to check in this pool controller for the rados namespace settings, right?

Reply:
Per offline discussion, the rados namespace controller is also configuring the mirroring on the pool, when the implicit namespace is reconciled. Let's see if we can implement this fix in that controller instead.